client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -150,14 +150,28 @@ public boolean hasNext() {
     return recordsIterator.hasNext();
   }
 
+  private boolean isSameMemoryType(ByteBuffer left, ByteBuffer right) {
```

Comment:
how about `return left.isDirect() == right.isDirect()`?

Reply:
!left.isDirect() && !right.isDirect() is also OK

Reply:
`return left.isDirect() == right.isDirect()` includes the situation `!left.isDirect() && !right.isDirect()`.

Reply:
Oh my fault!

Reply:
haha